Create rule S6561: Do not use DateTime.Now for benchmarking or timing operations (#1711)
SonarSource/rspec: rules/S6561/why.adoc

> Using repeated calls to `DateTime.Now` to measure elapsed time is not recommended since this property is subject to changes such as DST transitions, which can cause issues if the change occurs during the benchmark session.

Reviewer: I'd avoid usings DST alone - as this way we assume that everyone knows what DST should mean.
I'd write it fully as daylight savings transition (or something like that).

> Using repeated calls to `DateTime.Now` to measure elapsed time is not recommended since this property is subject to changes such as DST transitions, which can cause issues if the change occurs during the benchmark session.

Reviewer 2: ```suggestion
Using repeated calls to `DateTime.Now` to measure elapsed time is not recommended since DST transition can happen between the `DateTime.Now` calls, thus invalidating the calculation.
```

Reviewer 3: I think the purpose of this wording was to highlight that DST transitions are one of many changes that can occur abruptly. But I like your sentence more so I'll see what I can do.

> Using repeated calls to `DateTime.Now` to measure elapsed time is not recommended since this property is subject to Daylight Savings Time (DST), which can cause issues if a DST transition occurs during the benchmark session.
> Moreover, `DateTime.Now` is dependent on the system clock, which may have low resolution on older systems (as low as 15 milliseconds).

Reviewer: ```suggestion
Using repeated calls to `DateTime.Now` to measure elapsed time is not recommended since this property is subject to changes such as Daylight Savings Time (DST), which can cause issues if the change occurs during the benchmark session.
Moreover, `DateTime.Now` is dependent on the system clock, which may have low resolution on older systems (as low as 15 milliseconds).
```
